### ISSUE TITLE
Proposal: Parallelize test suite runs

### ISSUE THREAD

**Issue by florianPat** [sebastianbergmann/phpunit#5093]
Hello phpunit-community,
I wanted to discuss the idea of running test cases in parallel to gain performance speedups for large test suites.
I created a "proof of conecpt" WIP implementation for my bachelors thesis, which uses threads to parallalsize the execution, using the php parallel extension.
The starting code to build upon or discuess the idea can be reviewed [here](https://github.com/florianPat/phpunit/commit/77728852d73ea4879eb02962c1485b4e47b0f91e)
I would love to see feedback on the idea and the pros and cons of the possibility to add this in the phpunit "core". I also evaluated the extension system, but did not find a way to hook into it to achieve what i intend.
I also know that the framework paratest exists, but in my test paratest was slower than phpunit without threading...

Thanks for your feedback,
Florian

**Comment by sebastianbergmann** [comment]
Until we have refactored how PHPUnit runs tests, I consider parallel test execution out of scope for this project. In the meantime, solutions such as [paratest](https://github.com/paratestphp/paratest) exist.
